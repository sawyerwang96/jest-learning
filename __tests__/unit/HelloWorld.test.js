import { shallowMount } from '@vue/test-utils'
// import Vue from 'vue'
import HelloWorld from '@/components/HelloWorld.vue'

describe('HelloWorld.vue', () => {
  // shallowMount只测试当前组件，不会测试子组件 适合单元测试
  // mount适合集成测试 测试组件间通信、关联时使用
  // it('renders props.msg when passed', () => {
  //   const msg = 'new message'
  //   const wrapper = shallowMount(HelloWorld, {
  //     propsData: { msg }
  //   })
  //   expect(wrapper.text()).toMatch(msg)
  // })

  // it('renders pops.msg when passed', () => {
  //   // 只能测试ui展示
  //   let root = document.createElement('div')
  //   root.className = 'root'
  //   document.body.appendChild(root)
  //   new Vue({
  //     render: h => h(HelloWorld, {
  //       props: {
  //         msg: 'wsy'
  //       }
  //     })
  //   }).$mount('.root')
  //   expect(document.getElementsByClassName('hello').length).toBe(1)
  // })

  it('component render', () => {
    const wrapper = shallowMount(HelloWorld, {
      props: {
        msg: 123
      }
    })
    // 快照能及时捕获ui更新内容
    expect(wrapper).toMatchSnapshot()
  })
})
