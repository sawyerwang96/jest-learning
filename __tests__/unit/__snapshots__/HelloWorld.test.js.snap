// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld.vue component render 1`] = `
<div class="hello">
  <h1></h1>
</div>
`;
